core/embed/bootloader/main.c

```diff
@@ -188,7 +188,15 @@ static usb_result_t bootloader_usb_loop(const vendor_header *const vhdr,
       case MessageType_MessageType_FirmwareUpload:
         r = process_msg_FirmwareUpload(USB_IFACE_NUM, msg_size, buf);
         if (r < 0 && r != UPLOAD_ERR_USER_ABORT) {  // error, but not user abort
+#ifdef USE_OPTIGA
+          if (r == UPLOAD_ERR_BOOTLOADER_LOCKED) {
+            ui_screen_install_restricted();
+          } else {
+            ui_screen_fail();
+          }
+#else
           ui_screen_fail();
+#endif
```

Comment:
let's avoid the ifdef here, and instead redefine `ui_screen_install_restricted` to `ui_screen_fail` if there is no optiga.

Reply:
https://github.com/trezor/trezor-firmware/commit/cc79e566602abcc4bb1a044dc6fb8d445a6051f3